buildSrc/src/main/minimumRuntime/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -19,21 +26,52 @@
 @SuppressWarnings("unchecked")
 public class LoggedExec extends Exec {
 
+    // helper to allow lambda creation using Files methods
+    private interface ThrowingStringSupplier {
+        String get() throws Exception;
+    }
+
+    private Property<Boolean> useFileBuffer;
+    
     public LoggedExec() {
-        ByteArrayOutputStream output = new ByteArrayOutputStream();
-        ByteArrayOutputStream error = new ByteArrayOutputStream();
+        this.useFileBuffer = getProject().getObjects().property(Boolean.class);
+        this.useFileBuffer.set(false); // default to in memory
```

Comment:
I believe the "convention" is to use `convention()` for setting default values on properties.